common/flatpak-installation.c

```diff
@@ -979,25 +979,51 @@ flatpak_installation_list_installed_refs_by_kind (FlatpakInstallation *self,
 }
 
 static gboolean
-ref_check_for_update (FlatpakDir   *dir,
-                      const char   *ref,
-                      GHashTable   *remote_states,
-                      GCancellable *cancellable)
+transaction_ready (FlatpakTransaction  *transaction,
+                   GHashTable         **related_to_refs)
 {
-  g_autoptr(GBytes) deploy_data = NULL;
-  FlatpakRemoteState *state;
-  const char *origin = NULL;
+  GList *ops = flatpak_transaction_get_operations (transaction);
+  GList *l;
 
-  deploy_data = flatpak_dir_get_deploy_data (dir, ref, FLATPAK_DEPLOY_VERSION_CURRENT, cancellable, NULL);
-  if (deploy_data == NULL)
-    return FALSE;
+  for (l = ops; l != NULL; l = l->next)
+    {
+      FlatpakTransactionOperation *op = l->data;
+      const char *ref = flatpak_transaction_operation_get_ref (op);
+      const char *related_to_ref = flatpak_transaction_operation_get_related_to_ref (op);
+      FlatpakTransactionOperationType type = flatpak_transaction_operation_get_operation_type (op);
+
+      /* There is currently no way for a set of updates to lead to an
+       * uninstall, but check anyway.
+       */
+      if (type == FLATPAK_TRANSACTION_OPERATION_UNINSTALL)
+        {
+          g_warning (_("Update transaction unexpectedly wants to uninstall %s"), ref);
+          continue;
+        }
 
-  origin = flatpak_deploy_data_get_origin (deploy_data);
-  state = g_hash_table_lookup (remote_states, origin);
-  if (state == NULL)
-    return FALSE;
+      if (related_to_ref == NULL)
+        g_warning (_("Transaction operation for %s unexpectedly not related to a ref"), ref);
```

Comment:
Same here.

Reply:
This should also `continue`, no?

Reply:
Or maybe we just assert that `related_to_ref != NULL`

Reply:
Removed these lines